packages/postgraphile-core/__tests__/kitchen-sink-schema.sql

```diff
@@ -1238,3 +1239,67 @@ create table geometry.geom (
   polygon polygon,
   circle circle
 );
+
+--------------------------------------------------------------------------------
+
+create schema function_returning_enum;
+
+create table function_returning_enum.stage_options (
+  type text primary key
+);
+comment on table function_returning_enum.stage_options is E'@enum';
+insert into function_returning_enum.stage_options (type) values ('pending'), ('round 1'), ('round 2'), ('rejected'), ('hired');
+
+create table function_returning_enum.applicants (
+  id int,
+  first_name text,
+  last_name text,
+  stage text references function_returning_enum.stage_options (type)
+);
+
+--- follow the convention of [enum_name]_enum_domain
+create domain function_returning_enum.stage_options_enum_domain as text;
+
+create or replace function function_returning_enum.applicants_next_stage(
+  a function_returning_enum.applicants
+) returns function_returning_enum.stage_options_enum_domain
+as $$
+  select (case when a.stage = 'round 2' then 'hired' 
+    else 'rejected' end)::function_returning_enum.stage_options_enum_domain;
+$$ language sql stable;
+comment on function function_returning_enum.applicants_next_stage is E'@filterable';
+
+create table function_returning_enum.length_status (
+  type text primary key
+);
+comment on table function_returning_enum.length_status is E'@enum';
+insert into function_returning_enum.length_status (type) values ('ok'), ('too_short');
+
+--- use an @enum tag this time
+create domain function_returning_enum.length as text;
+comment on domain function_returning_enum.length is E'@enum length_status';
+
+create or replace function function_returning_enum.applicants_name_length(
+  a function_returning_enum.applicants
+) returns function_returning_enum.length
+as $$
+  select (case when length(a.last_name) < 4 then 'too_short' 
+    else 'ok' end)::function_returning_enum.length;
+$$ language sql stable;
+
+create or replace function function_returning_enum.text_length(
+  text text,
+  min_length int
+) returns function_returning_enum.length
+as $$
+  select (case when length(text) < min_length then 'too_short' 
+    else 'ok' end)::function_returning_enum.length;
+$$ language sql stable;
+
+create or replace function function_returning_enum.applicants_name_length(
+  a function_returning_enum.applicants
+) returns function_returning_enum.length
+as $$
+  select function_returning_enum.text_length(a.last_name, 4)::function_returning_enum.length;
+$$ language sql stable;
+comment on function function_returning_enum.applicants_name_length is E'@filterable';
```

Comment:
These are all great, but we also need to see usage where the enum domain is used as an argument to the function (rather than the return type as above).